core/src/main/java/tech/tablesaw/io/Source.java

```diff
@@ -84,12 +106,37 @@ public Reader createReader(byte[] cachedBytes) throws IOException {
             return new InputStreamReader(new ByteArrayInputStream(cachedBytes));
         }
         if (inputStream != null) {
-            return new InputStreamReader(inputStream, charset);
+            return getReader(inputStream);
         }
         if (reader != null) {
             return reader;
         }
-        return new InputStreamReader(new FileInputStream(file), charset);
+        return getReader(new FileInputStream(file));
+    }
+
+    /**
+     *  Returns a new Reader on the underlying source object.
+     */
+    private Reader getReader(InputStream raw) throws IOException {
+        // we want to remove the BOM windows adds from a file if the encoding is UTF-8
+        // in other cases we depend on the charsets
+        InputStream in = new BufferedInputStream(raw);
+        if (UTF8.name().equalsIgnoreCase(charset.name())) {
+            in.mark(3);
+            boolean hasBOM = true;
```

Comment:
We use univocity-parsers, which is supposed to do BOM handling for us. It's not clear to me why this would be required. Can we either change our test or file a bug with univocity so that we don't have to do any special logic here?

Reply:
Okay, I didn't know about BOM handling within univocity-parsers. It seems like you need to explicitly make use of the BomInput class for univocity-parsers to do what is needed. I'll see if I can rework the PR based around that.